common/scala/src/main/scala/whisk/core/database/cosmosdb/CosmosDBArtifactStore.scala

```diff
@@ -206,6 +217,10 @@ class CosmosDBArtifactStore[DocumentAbstraction <: DocumentSerializer](protected
       RxReactiveStreams.toPublisher(client.queryDocuments(collection.getSelfLink, querySpec, newFeedOptions()))
     val f = Source
       .fromPublisher(publisher)
+      .map { f =>
+        collectMetrics(queryToken, f.getRequestCharge)
+        f
+      }
```

Comment:
Could you use `wireTap` here?

Reply:
Yes `wireTap` is better operator to use. Switched to that now